src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -264,7 +295,7 @@ CollectedHeap::CollectedHeap() :
   NOT_PRODUCT(_promotion_failure_alot_count = 0;)
   NOT_PRODUCT(_promotion_failure_alot_gc_number = 0;)
 
-  if (UsePerfData) {
+  if (UsePerfData && os::is_thread_cpu_time_supported()) {
```

Comment:
This condition should be a nested if inside `if (UsePerfData)`:

```
if (os::is_thread_cpu_time_supported()) {
  _total_cpu_time = ...;
  _perf_parallel_worker_threads_cpu_time = ...;
}
```
Otherwise `_perf_gc_cause` and `_perf_gc_lastcause` could be broken.

Reply:
Ah yes, good catch!